File: ios/Classes/Data/ErrorCodes.swift
// Copyright © 2022 Olo Inc. All rights reserved.
// This software is made available under the Olo Pay SDK License (See LICENSE.md file)
//
//  ErrorCodes.swift
//  olo_pay_sdk
//
//  Created by Justin Anderson on 12/8/23.
//

import Foundation

public class ErrorCodes {
    // Method Call Rejection Codes
    public static let InvalidParameter = "InvalidParameter"
    public static let MissingParameter = "MissingParameter"
    public static let UnexpectedParameterType = "UnexpectedParameterType"
    public static let UninitializedSdk = "SdkUninitialized"
    public static let ApplePayUnsupported = "ApplePayUnsupported"
    public static let Unimplemented = "UNIMPLEMENTED"
    public static let ViewNotFound = "ViewNotFound"
    
    // Do not rename this key... it maps to an OPError type, but we also
    // use if for some other scenarios in the RN SDK
    public static let GeneralError = "generalError"
    
    // API Exception Error Codes
    public static let ApiError = "ApiError"
    public static let InvalidRequest = "InvalidRequest"
    public static let Connection = "ConnectionError"
    public static let Cancellation = "CancellationError"
    public static let Authentication = "AuthenticationError"
    
    // Card Exception Error Codes
    public static let InvalidCardDetails = "InvalidCardDetails"
    public static let InvalidNumber = "InvalidNumber"
    public static let InvalidExpiration = "InvalidExpiration"
    public static let InvalidCvv = "InvalidCVV"
    public static let InvalidPostalCode = "InvalidPostalCode"
    public static let ExpiredCard = "ExpiredCard"
    public static let CardDeclined = "CardDeclined"
    public static let ProcessingError = "ProcessingError"
    public static let UnknownCard = "UnknownCardError"
    
    // Font Error Codes
    public static let UnexpectedError = "UnexpectedError"
    public static let AssetNotFoundError = "AssetNotFoundError"
    public static let FontLoadError = "FontLoadError"
}

File: android/src/main/kotlin/com/olo/flutter/olo_pay_sdk/data/ErrorCodes.kt
// Copyright © 2022 Olo Inc. All rights reserved.
// This software is made available under the Olo Pay SDK License (See LICENSE.md file)
package com.olo.flutter.olo_pay_sdk.data

class ErrorCodes {
    companion object {
        // Method call rejection code keys
        const val InvalidParameter = "InvalidParameter"
        const val MissingParameter = "MissingParameter"
        const val UnexpectedParameterType = "UnexpectedParameterType"
        const val UninitializedSdk = "SdkUninitialized"
        const val DigitalWalletUninitialized = "DigitalWalletUninitialized"
        const val InvalidGooglePaySetup = "InvalidGooglePaySetup"
        const val DigitalWalletNotReady = "DigitalWalletNotReady"
        const val UnexpectedError = "UnexpectedError"

        // Do not rename this key... it maps to an OPError type, but we also
        // use if for some other scenarios in the RN SDK
        const val GeneralError = "generalError"

        // Error codes based on API exceptions
        const val ApiError = "ApiError"
        const val InvalidRequest = "InvalidRequest"
        const val Connection = "ConnectionError"
        const val RateLimit = "RateLimitError"
        const val Authentication = "AuthenticationError"

        // Error codes base on card exceptions
        const val InvalidCardDetails = "InvalidCardDetails"
        const val InvalidNumber = "InvalidNumber"
        const val InvalidExpiration = "InvalidExpiration"
        const val InvalidCvv = "InvalidCVV"
        const val InvalidPostalCode = "InvalidPostalCode"
        const val ExpiredCard = "ExpiredCard"
        const val CardDeclined = "CardDeclined"
        const val ProcessingError = "ProcessingError"
        const val UnknownCardError = "UnknownCardError"
    }
}
